### PR TITLE
WiP: Install omero

### PR DESCRIPTION
**Pull request by pwalczysko**
WiP: As discussed last week with @jburel , adding a workflow guide for OMERO install, based freely on the presentation https://downloads.openmicroscopy.org/presentations/2020/Dundee/Workshops/OME2020-OMERO-Installation/#/

See also 
https://github.com/ome/omero-deployment-examples/pull/70
https://github.com/ome/ansible-example-omero-onenode/pull/11

Rebased this on top of https://github.com/ome/omero-guides/pull/236 to avoid scc conflitcs

### CODE REVIEW

**Comment by pwalczysko** [comment]
Interestingly, for https://github.com/ome/omero-guides/pull/232/commits/e5e4ba257bf015f9355e3f680f7aef2eef633d4a (expose OMERO.web in Ansible install case on CentOS 7 docker) I had to add to the web config inside the docker:

```
omero config set omero.web.application_server.host 0.0.0.0
```
This finally exposed the OMERO.web outside of the docker on localhost:4080. I wonder if the completely empty config with respect to OMERO.web after installing omero using Ansible as per [playbook in ansible..-onenode example](https://github.com/ome/ansible-example-omero-onenode/blob/master/playbook.yml) is not a bug of the workflow. We probably should not claim it works out of the box in OMERO.web case for ``onenode``, without an additional config ?
Even though I added the line above to the config manually, I still do not have correct css on the page - any idea what should be yet tweaked to have full experience @will-moore please ?
Screenshot 2022-08-24 at 18 34 59

File: install.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+-  Where to find the OMERO.server requirements.
+
+-  How to install OMERO.server quickly using a simple Ansible playbook example.
+
+-  How to install OMERO.server very quickly using a provided Docker image.
+
+-  How to configure OMERO.server to work with LDAP, both using Ansible or using manual configuration.
+
+-  How to install apllications for OMERO.web, such as OMERO.figure.
```
Comment (will-moore) [inline install.rst:20]: typo: apllications

File: install.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+Clone the subrepository for the one-node example::
+
+    $ git clone https://github.com/ome/ansible-example-omero-onenode.git
+    $ cd ansible-example-omero-onenode	
+
+Install requrements::
+
+    $ ansible-galaxy install -r requirements.yml
+
+Creat a file ``hosts.yml`` in the same directory and put into it (replacing the YOUR-HOST-NAME variable as appropriate)::
```
Comment (will-moore) [inline install.rst:61]: typo Creat

Comment (will-moore) [reply install.rst:61]: What should `YOUR-HOST-NAME` be here? `wills_server`? Anything I like?

File: install.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      hosts:
+        YOUR-HOST-NAME 
+
+Edit the ``playbook.yml`` file to change the OMERO root password (by default ``ChangeMe``). Then use this playbook to install the OMERO.server and OMERO.web on one node::
+
+    $ ansible-playbook --become -i hosts.yml playbook.yml
+
+Start OMERO.insight or a Command Line Interface (CLI) and log in to ``localhost`` with the username ``root`` and the password you have set in the ``playbook.yml`` above.
+
+Go to ``localhost/4080`` in your browser and log in to OMERO.web with the same credentials you used for OMERO.insight above.
```
Comment (will-moore) [inline install.rst:73]: Should be: http://localhost:4080/ and probably nicer to make it into a clickable link

File: install.rst [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+Docker
+------
+
+Even quicker option than ansible is docker, but you have to be comfortable working with containers.
+
+Assuming that the Docker is installed, clone the deployment examples repository (if not already cloned)::
```
Comment (will-moore) [inline install.rst:80]: remove 'the' -> "Assuming that Docker is installed"

File: install.rst [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    $ git clone https://github.com/ome/docker-example-omero.git
+    $ cd docker-example-omero
+
+Pull the latest versions of the containers and start them::
+
+    $ docker-compose pull
+    $ docker-compose up -d
+    $ docker-compose logs -f
+
+Go to ``0.0.0.0/4080`` in your browser and log in to OMERO.web as ``root`` with password ``omero``.
```
Comment (will-moore) [inline install.rst:93]: As above, clickable link is nice.

File: install.rst [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+Start OMERO.insight or a Command Line Interface (CLI) and log in to ``localhost`` with the same credentials.
+
+
+
+You also have the option to follow the server installation steps at https://omero.readthedocs.io/en/stable/sysadmins/unix/server-installation.html or 
+But you will have to install apps and set up the config according to the ansible playbook anyway.
```
Comment (will-moore) [inline install.rst:100]: "according to the ansible playbook" - Will users know what that means? I guess you have to look in the right place in one of the repos listed above, but we must have a better resource for how to install web apps?

File: install.rst [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    $ docker-compose up -d
+    $ docker-compose logs -f
+
+Go to ``0.0.0.0/4080`` in your browser and log in to OMERO.web as ``root`` with password ``omero``.
+
+Start OMERO.insight or a Command Line Interface (CLI) and log in to ``localhost`` with the same credentials.
+
+
+
+You also have the option to follow the server installation steps at https://omero.readthedocs.io/en/stable/sysadmins/unix/server-installation.html or
```
Comment (will-moore) [inline install.rst:99]: line ends with " or ..."

**Comment by will-moore** [comment]
@pwalczysko Sorry, in answer to your question, I don't know what's going on or where to look. I guess the code that's responsible is in https://github.com/ome/ansible-role-omero-web somewhere but I've not used ansible yet or edited a playbook/role.

File: install.rst (hunk 2 above)
Comment (pwalczysko) [reply install.rst:61]: Definitely not anything you like. Something like https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html (look for "hosts" in the examples). This is not trivial, I will add some formatting examples to the text as you suggest. Ultimately, it is very similar (but not the same) to "enter here your OMERO.server address" - the same level of difficulty to explain the "non-triviality". Also, having a valid host on which I can install OMERO at all is a difficult step...

Comment (pwalczysko) [reply install.rst:61]: Actually, I mention in the steps above "prepare an empty box with CentOS 7 on it", I will refer to it here.

File: install.rst (hunk 6 above)
Comment (pwalczysko) [reply install.rst:100]: Thank you, this is very much WiP yet - needs a complete rehaul

**Comment by joshmoore** [comment]
Can someone explain the difference between this new page and the main installation document?

**Comment by pwalczysko** [comment]
> Can someone explain the difference between this new page and the main installation document?

The idea was to have a very simple quick walkthrough, something based on https://downloads.openmicroscopy.org/presentations/2020/Dundee/Workshops/OME2020-OMERO-Installation/#/ , which can be used in workshop situations (this PR). Target audience is a sysadmin who wants to have a quick chack, look and feel about what OMERO install entails.
 It also introduces the ``examples`` (ansible examples, docker examples) and makes a point recommending the usage of ansible and lets the sysadmin orientate in the possibilities and weigh pros and cons.

The main installation document is for sysadmins who already know that they will want to use manual install and not ansible, and want to go "the hard way" as explained in https://downloads.openmicroscopy.org/presentations/2020/Dundee/Workshops/OME2020-OMERO-Installation/#/ - much less streamlined, much more work, much less explanation about options.

**Comment by joshmoore** [comment]
Other than the "quick check" which I can see in terms of an intro before a sysadmin wants to jump into the deep-end (i.e. what are my options), I'd raise the question whether `ansible` shouldn't be an option beside the other manual installations. Why would that live in another place?

**Comment by jburel** [comment]
We discussed several times adding the ``ansible`` instructions to the main documentation (which I have been for) along side the "step-by-step" ones but we never reached a consensus.

**Comment by joshmoore** [comment]
@pwalczysko: thoughts?

**Comment by pwalczysko** [comment]
> @pwalczysko: thoughts?

I am using the work to get more familiar with the ansible ecosystem and to be able to answer the questions of sysadmins at outreach opportunities. The guide would be kind of a necessary first step, as it will be necessarily minimal. It also would be having the overview of options (ansible, docker, manual) in a much more lighweight manner than the serious doc and would be an easy link to point to when the installation questions come (they do not necessarily come from sysadmins, but from men-in-the-middle).

So I would vote for both, first this walkthrough, and when I see that [it works](https://github.com/ome/ansible-example-omero-onenode/pull/16#issuecomment-1240608491) and makes sense in the overall and simple cases, then inclusion in the main doc.

File: install.rst [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+    $ git clone https://github.com/ome/omero-deployment-examples.git
+    $ cd omero-deployment-examples
+    $ git clone https://github.com/ome/docker-example-omero.git
+    $ cd docker-example-omero
+
+Pull the latest versions of the containers and start them::
+
+    $ docker-compose pull
+    $ docker-compose up -d
+    $ docker-compose logs -f
```
Comment (will-moore) [inline install.rst:87]: This works so far... and gets to:
```
omeroserver_1  | 2022-09-22 13:08:48,441 [    omego.db] INFO  Database is already at OMERO5.4__0
omeroserver_1  | Running /startup/99-run.sh 
omeroserver_1  | Starting OMERO.server
```
but then seems to stop.
Webclient is available at http://localhost:4080/webclient but says server is not responding.
How do I access server logs to see if anything went wrong?

Comment (will-moore) [reply install.rst:87]: I have:
```
$ docker ps
CONTAINER ID   IMAGE                                   COMMAND                  CREATED      STATUS         PORTS                              NAMES
4e63dfd4de54   openmicroscopy/omero-server:5           "/usr/local/bin/entr…"   5 days ago   Up 8 minutes   0.0.0.0:4063-4064->4063-4064/tcp   docker-example-omero_omeroserver_1
7bb7d25fb5f7   openmicroscopy/omero-web-standalone:5   "/usr/local/bin/entr…"   5 days ago   Up 8 minutes   0.0.0.0:4080->4080/tcp             docker-example-omero_omeroweb_1
acb5f42fa4cf   postgres:11                             "docker-entrypoint.s…"   5 days ago   Up 8 minutes   5432/tcp                           docker-example-omero_database_1
```

Comment (will-moore) [reply install.rst:87]: OK, so I managed to ssh into server...

```
$ docker exec -it docker-example-omero_omeroserver_1 /bin/bash
```
I don't know where `$OMERO_HOME` is, thought maybe I could find something like `grid.xml`...

```
bash-4.2$ cd /
bash-4.2$ find ./ -name grid.xml
find: './var/spool/at': Permission denied
find: './var/spool/cron': Permission denied
find: './var/db/sudo': Permission denied
find: './var/lib/machines': Permission denied
find: './var/lib/yum/history/2020-11-13/2': Permission denied
find: './var/lib/yum/history/2020-11-13/1': Permission denied
find: './var/lib/yum/history/2020-11-13/8': Permission denied
find: './var/lib/yum/history/2020-11-13/11': Permission denied
find: './var/lib/yum/history/2020-11-13/7': Permission denied
find: './var/lib/yum/history/2020-11-13/9': Permission denied
find: './var/lib/yum/history/2020-11-13/6': Permission denied
find: './var/lib/yum/history/2020-11-13/10': Permission denied
find: './var/lib/yum/history/2020-11-13/5': Permission denied
find: './var/lib/yum/history/2020-11-13/3': Permission denied
find: './var/lib/yum/history/2020-11-13/4': Permission denied
find: './etc/pki/CA/private': Permission denied
find: './etc/ssmtp': Permission denied
find: './etc/sudoers.d': Permission denied
find: './proc/tty/driver': Permission denied
find: './run/cryptsetup': Permission denied
find: './run/sudo': Permission denied
find: './root': Permission denied

bash-4.2$ sudo find ./ -name grid.xml
sudo: no tty present and no askpass program specified

bash-4.2$ echo $OMERO_HOME

```

Comment (will-moore) [reply install.rst:87]: Thanks - using my `exec` still on omero-server container... This gives me nothing, so something failing before logs created:
```
ls /opt/omero/server/OMERO.server/var/
```

Comment (pwalczysko) [reply install.rst:87]: > I have:
> 
> ```
> $ docker ps
> CONTAINER ID   IMAGE                                   COMMAND                  CREATED      STATUS         PORTS                              NAMES
> 4e63dfd4de54   openmicroscopy/omero-server:5           "/usr/local/bin/entr…"   5 days ago   Up 8 minutes   0.0.0.0:4063-4064->4063-4064/tcp   docker-example-omero_omeroserver_1
> 7bb7d25fb5f7   openmicroscopy/omero-web-standalone:5   "/usr/local/bin/entr…"   5 days ago   Up 8 minutes   0.0.0.0:4080->4080/tcp             docker-example-omero_omeroweb_1
> acb5f42fa4cf   postgres:11                             "docker-entrypoint.s…"   5 days ago   Up 8 minutes   5432/tcp                           docker-example-omero_database_1
> ```

vs mine which is  (I have some additional :: motives in the PORTS bit, but not sure how important this is)

```
docker ps
CONTAINER ID   IMAGE                                   COMMAND                  CREATED          STATUS          PORTS                                                           NAMES
3d049e9f93a0   postgres:11                             "docker-entrypoint.s…"   35 seconds ago   Up 34 seconds   5432/tcp                                                        docker-example-omero_database_1
8fd37b70d98b   openmicroscopy/omero-web-standalone:5   "/usr/local/bin/entr…"   35 seconds ago   Up 34 seconds   0.0.0.0:4080->4080/tcp, :::4080->4080/tcp                       docker-example-omero_omeroweb_1
3e3b742adb6d   openmicroscopy/omero-server:5           "/usr/local/bin/entr…"   35 seconds ago   Up 34 seconds   0.0.0.0:4063-4064->4063-4064/tcp, :::4063-4064->4063-4064/tcp   docker-example-omero_omeroserver_1
```

Comment (pwalczysko) [reply install.rst:87]: @will-moore I have master.err and Blitz log, but my server is not working on M1. I will open an extra issue, so as not to confuse people.

Comment (pwalczysko) [reply install.rst:87]: @will-moore created https://github.com/ome/docker-example-omero/issues/12

**Comment by joshmoore** [comment]
When would be the right time to come back to the discussion from early Sep?

**Comment by will-moore** [comment]
I am still finding it quite painful to do any work that needs a connection to OMERO, especially when working from home, since I still have not managed to install OMERO on my M1 Mac, even using Docker etc.

So I would really appreciate *any* solution that allows me to run OMERO locally!

**Comment by pwalczysko** [comment]
> When would be the right time to come back to the discussion from early Sep?

The sooner the better for me. cc @jburel - I should be finishing this by the end of January

**Comment by pwalczysko** [comment]
> I am still finding it quite painful to do any work that needs a connection to OMERO, especially when working from home, since I still have not managed to install OMERO on my M1 Mac, even using Docker etc.
> 
> So I would really appreciate _any_ solution that allows me to run OMERO locally!

@will-moore the install of OMERO on M1 is on my ToDo list prior to TiM 2023. I have my old Mac with everything running on it, but the M1 machine just became my main working comp. I am afraid though that we had a discrepancy in the experience of spinning up the docker between mine M1 and yours. No idea why.  Do you want to please rather discuss on a separate issue on the https://github.com/ome/ansible-example-omero-onenode rather than this PR please ?

**Pull request closed without merging**